test/topology_custom/test_maintenance_mode.py

```diff
@@ -88,6 +91,8 @@ async def test_maintenance_mode(manager: ManagerClient):
     # Restart in normal mode to see if the changes made in maintenance mode are persisted
     await manager.server_update_config(server_a.server_id, "maintenance_mode", "false")
     await manager.server_start(server_a.server_id, wait_others=1)
+    await wait_for_cql_and_get_hosts(cql, [server_a], time.time() + 60)
```

Comment:
Maybe we need to mention in `server_start()` (`manager_client.py`) that `server_start()` will only wait for the node to finish initializing and start listening on the CQL port, it does *not* wait for the other nodes to see it as UP.

Although, I imagine most people copy-paste these snippets from some other test and they won't see the comment. Still, it wouldn't hurt.

Reply:
Alternative: dtest has `wait_other_notice` parameter in their version of `server_start`, if `True` the function will wait for other servers to notice the restarted server as UP.

That would get rid of the need for explicit `servers_see_each_other`

And we could do similar for the CQL wait; we could have a `wait_for_cql` parameter, which would get a `Session | None` object. The default would be `get_cql()` (from within `ManagerClient`), but test could also pass `None` (if it cannot have a session at this point for some reason) or some other session if it needs to use one.

Reply:
Even better, the less boilerplate, the better (we could leave the current exploded methods too, for a test which needs more fine-grained control).